src/pre_processors/nltk_tokenization_and_stopword_removal.py

```diff
@@ -0,0 +1,37 @@
+from typing import Union, List
+
+import nltk
+from nltk import word_tokenize
+from nltk.corpus import stopwords
+
+from arqmath_code.Entities.Post import Question, Answer
+from arqmath_code.topic_file_reader import Topic
+from src.base.pre_processing import PreProcessor
+from src.base.query_pre_processor import QueryPreProcessor
+
+nltk.download('stopwords')
+nltk.download('punkt')
+stop_word_set = set(stopwords.words('english'))
+
+
+def tokenize_text(text: str) -> str:
+    text_tokens = word_tokenize(text)
+    tokens_without_sw = [word for word in text_tokens if not word in stop_word_set]
```

Comment:
If you really want to remove all stop words, you have to make a lower case conversion to the target string, as the stop words in nltk are only in lowercase. So you'll miss Uppercase stop words here

Reply:
fixed